csrc/device_lower/pass/index.cpp

```diff
@@ -1507,16 +1507,41 @@ void IndexLowering::handle(const LoadStoreOp* ldst) {
           std::make_shared<DataType>(DataType::UInt32),
           (size_t)ir_utils::getVectorizeSize(ldst->out()->as<TensorView>()) /
               2};
+    } else if (ir_utils::isStMatrixOp(ldst)) {
+      as_type = ArrayType{
+          std::make_shared<DataType>(DataType::UInt32),
+          (size_t)ir_utils::getVectorizeSize(ldst->in()->as<TensorView>()) / 2};
```

Comment:
I believe I understand why you write this part like this: you need to know the load size to determine whether it is `.x1`, `.x2`, or `.x4`. But I don't think the code here is correct. What we are checking here is the vectorization factor of `ldst->in()`, which is how we write the register tensor, instead of how we read register. What we actually do is to check the consumer's vectorization factor, for this case, the smem tv's vectorization factor. But unfortunately, in this PR, we are not scheduling it at all. How about we just hardcode this value again, instead of using a wrong value?